rest/server_context.go

```diff
@@ -1006,6 +1008,7 @@ func (sc *ServerContext) startStatsLogger() {
 func (sc *ServerContext) stopStatsLogger() {
 	if sc.statsContext.terminator != nil {
 		close(sc.statsContext.terminator)
+		sc.statsContext.statsLoggerWg.Wait()
```

Comment:
Why did you choose to use a waitGroup here, instead of the done channel approach used for background tasks?

Reply:
The reason that I’ve used WaitGroup here was that I didn’t take timeout handling into consideration initially. But, this has been refactored with channel based wait similar to what we’ve in background tasks termination handling.